src/units/KM_Units.pas

```diff
@@ -307,6 +313,79 @@ function TKMUnitCitizen.FindHome: Boolean;
 end;
 
 
+procedure TKMSettledUnit.CleanHousePointer(aFreeAndNilTask: Boolean = False);
+begin
+  if aFreeAndNilTask then
+    FreeAndNil(fUnitTask);
+  fHome.GetHasOwner := False;
+  gHands.CleanUpHousePointer(fHome);
+end;
+
+// Manage house is closed for worker process
+procedure TKMSettledUnit.ProceedHouseClosedForWorker;

```

Comment:
Ok, I'll check it out, thx!
This feature was my very first in KaM Remake, so I probably missed something in October '16 :)